onchain/rollups/contracts/consensus/quorum/Quorum.sol

```diff
@@ -12,48 +13,47 @@ import {IHistory} from "../../history/IHistory.sol";
 /// @title Quorum consensus
 /// @notice A consensus model controlled by a small set of addresses, the validators.
 ///         In this version, the validator set is immutable.
-/// @dev This contract uses OpenZeppelin `PaymentSplitter`.
-///      For more information on `PaymentSplitter`, please consult OpenZeppelin's official documentation.
+/// @dev This contract uses OpenZeppelin `PaymentSplitter` and `BitMaps`.
+///      For more information on those, please consult OpenZeppelin's official documentation.
 contract Quorum is AbstractConsensus, PaymentSplitter {
+    using BitMaps for BitMaps.BitMap;
+
     /// @notice The history contract.
     /// @dev See the `getHistory` function.
     IHistory internal immutable history;
 
     // Quorum members
-    // Map an address to true if it's a validator
-    mapping(address => bool) public validators;
+    // Map an address to its index in the validator set.
+    // The first validator has index 1. Thus, index 0 means the address is not in the validator set.
+    mapping(address => uint256) public validatorIndex;
     uint256 public immutable quorumSize;
 
     // Quorum votes
     struct Voted {
+        // how many has voted yea
         uint256 count;
-        // Map an address to true if it has voted yea
-        mapping(address => bool) voted;
+        // use BitMap to record who has voted
+        BitMaps.BitMap votedBitMap;
     }
     // Map a claim to struct Voted
     mapping(bytes => Voted) internal yeas;
 
     /// @notice Raised if not a validator
     error OnlyValidator();
-    modifier onlyValidator() {
```

Comment:
Hey! Just a question, why is it better to throw an error than just defining a modifier?

Reply:
I meant why did you change the modifier implementation in favor of de error?

Reply:
Oh, I see what you mean now.
It was to avoid reading from the same storage slot twice.
See the thread in https://github.com/cartesi/rollups-contracts/pull/57#discussion_r1334915237.

Reply:
Good catch! Thanks